qemu/tests/thin_provisioning.py

```diff
@@ -0,0 +1,176 @@
+import logging
+import os
+import re
+
+from autotest.client import os_dep
+from autotest.client.shared import utils
+
+from avocado.core import exceptions
+from avocado.utils import process
+
+from virttest import env_process
+from virttest import error_context
+
+
+def get_scsi_disk(session=None):
+    """
+    Get latest scsi disk which emulated by scsi_debug module.
+
+    :param session: Guest session
+    :return: Scsi id and the disk name
+    """
+    if session is None:
+        scsi_disk_info = process.system_output("lsscsi", shell=True).splitlines()
+        scsi_debug = [_ for _ in scsi_disk_info if 'scsi_debug' in _][-1]
+    else:
+        scsi_disk_info = session.cmd_output('lsscsi').splitlines()
+        scsi_debug = [_ for _ in scsi_disk_info][-1]
+    scsi_debug = scsi_debug.split()
+    scsi_id = scsi_debug[0][1:-1]
+    device_name = scsi_debug[-1]
+    return scsi_id, device_name
+
+
+def get_provisioning_mode(pci_id, device, session=None):
+    """
+    Get disk provisioning_mode, value usually is 'writesame_16', depends
+    on params for scsi_debug module.
+
+    :param pci_id: Pci slot number
+    :param device: The disk name
+    :param session: Guest session
+    :return:
+    """
+    device_name = os.path.basename(device)
+    path = "/sys/block/{:s}/device/scsi_disk".format(device_name)
+    path += "/{:s}/provisioning_mode".format(pci_id)
+    try:
+        if session is None:
+            o = utils.read_one_line(path).strip()
+            return o
+        else:
+            o = session.cmd_output("cat %s" % path)
+            return re.search(r"(unmap)", o).group(1)
```

Comment:
looks,  in latest kernel think_provisioning is not way writesame or umap.
[xutian@localhost staf-kvm-devel]$ cat /sys/block/sda/device/scsi_disk/1\:0\:0\:0/thin_provisioning 
1